### PR TITLE
new computation of variance according to XueLinSunSong 2018

### PR DESCRIPTION
**Pull request by rdanitz**

### CODE REVIEW

File: src/MarkovChain.hs [hunk 1]
```diff
@@ -96,25 +98,25 @@ norm_F = sqrt . sum . toList . fmap (^(2 :: Int))
 fundamental
   :: M   -- ^ Reduced matrix (n x n).
   -> M   -- ^ n x n
-fundamental q = case inverse (identity (nrows q) .- q) of
+fundamental m = case inverse (identity (nrows m) .- m) of
```
Comment (stevana) [inline src/MarkovChain.hs]: Why? Isn't the reduced matrix usually called `q`?

File: src/MarkovChain.hs [hunk 2]
```diff
@@ -206,49 +208,63 @@ expectedLength = norm_1 . rowVector . (getRow 1)
...
 -- │ 0.9523809523809523 0.9090909090909091 │
 -- │ 0.8695652173913043 0.8333333333333334 │
 -- └                                       ┘
-successRate
+expectedArcReliability
   :: Maybe (M, M)
   -> (M, M)
-  -> M
-successRate mprior (obsSuccs, obsFails) = succs ./ (succs + fails)
+  -> (M, M) -- ^ (mean, variance)
```
Comment (stevana) [inline src/MarkovChain.hs]: You need to update the doctests.

File: src/MarkovChain.hs [hunk 3]
```diff
@@ -259,13 +275,30 @@ transientReliability q mprior obs = rstar
     rstar :: M -- n-1 x 1
     rstar = fundamental fancyRdot * w
 
+    r2 :: M
+    r2 = r1 .* r1 + var
```
Comment (stevana) [inline src/MarkovChain.hs]: Does `.*` bind stronger than `+`? We should probably give the custom infix operators the appropriate precedence.

File: src/MarkovChain.hs (hunk 1 above)
Comment (rdanitz) [reply src/MarkovChain.hs]: Fundamental matrix is a more – well – fundamental concept, not only with regards to usage matrix. E. g. on :276 you can see `fundamental` being computed for `fancyRdot`. This confusion prompted me to change the arg name in the first place.

**Pull request merged**